### ISSUE TITLE
Feature request: Travis PHP min/max version

### ISSUE THREAD

**Issue by vanderlee** [badges/shields#819]
travis.yml files for PHP project contain a list of versions which are tested.
It would be useful to have a shield listing the minimum and/or maximum PHP version tested.

I'm assuming something similar will work for other programming languages testable with TravisCI.

**Comment by peter-gribanov** [comment]
I do not think that the maximum version is needed. The minimum is enough.

It might look like this:

![Minimum PHP Version](https://img.shields.io/badge/php-%3E%3D_5.5.9-8892BF.svg)

Markdown:

```
[![Minimum PHP Version](https://img.shields.io/badge/php-%3E%3D_5.5.9-8892BF.svg)](https://github.com/symfony/symfony)
```

You can get this info from [composer.json](https://github.com/symfony/symfony/blob/d8594de9c64d3f84fd4c27053a9f7b06d3cb47e0/composer.json#L19) or from [.travis.yml](https://github.com/symfony/symfony/blob/a84f59e644232e9909db0acfb9c6292f2b597ffd/.travis.yml#L27).

**Comment by vanderlee** [comment]
Maximum version is useful to indicate libraries that may be incompatible with newer versions of PHP (i.e. 7.x). It helps with transitioning existing applications from 5.x to 7.x, as 7.x introduced some significant backwards incompatibilities.

[![Compatible PHP Version](https://img.shields.io/badge/php-5.3%20--%207.0,%20HHVM-8892BF.svg)](https://github.com/symfony/symfony)

**Comment by peter-gribanov** [comment]
@vanderlee  Okay. I agree.
If the library is not tested on the PHP 7.1, then it probably does not support this version.

Most likely the library will work, because the developers of PHP do not like to break BC. But there are no guarantees for this. Guarantee can only be a running test and the running tests are determined by Travis CI.

**Comment by peter-gribanov** [comment]
> p.s. I see no reason why we couldn't combine both.

I agree.

Just in case, i will explain to others.
If we take data only from `.travis.yml`, then we can say on which versions of the PHP the library was tested.
If tested on `5.4`, `5.5`, `5.6` then we can write `5.4 - 5.6`.
If tested on `5.6`, `7.0` then we can write `5.6 - 7.0`.
If tested on `5.6`, `7.0`, `7.1` and PHP last version is `7.1` then we can write `>= 5.6`.
If tested only on `5.4` and `5.6` then we can write `5.4, 5.6`.

This is a complication of the rules, but it guarantees the accuracy and reliability of the data.

If there is no file `.travis.yml`, but there is a file ` composer.json` and in it the directive `php` is specified, it is possible to use it. This data is not reliable for libraries that are not updated regularly, but for lack of other data, you can use these.

**Comment by peter-gribanov** [comment]
I found solution for this. Is a [PHP-Eye](https://php-eye.com/) service.

https://php-eye.com/package/symfony/symfony
https://php-eye.com/api/v1/package/symfony/symfony.json

**Comment by peter-gribanov** [comment]
You can take information from composer.json with Packagist API

Example for [Symfony](https://github.com/symfony/symfony):
https://packagist.org/p/symfony/symfony.json

![image](https://user-images.githubusercontent.com/1954436/27383298-0be40c54-5693-11e7-82c6-94bd8f8856ef.png)

Docs
https://packagist.org/apidoc#get-package-by-name

**Comment by peter-gribanov** [comment]
Or can get info from [Travis CI API](https://docs.travis-ci.com/api/).

Example for [Symfony 3.4](https://github.com/symfony/symfony):

```
GET /repos/symfony/symfony/branches/3.4 HTTP/1.1
User-Agent: MyClient/1.0.0
Accept: application/vnd.travis-ci.2+json
Host: api.travis-ci.org
```
![image](https://user-images.githubusercontent.com/1954436/28167934-13e0a336-67e6-11e7-8f4f-e4bafbe8d246.png)

and then get info about job by id

```
GET /jobs/253120209 HTTP/1.1
User-Agent: MyClient/1.0.0
Accept: application/vnd.travis-ci.2+json
Host: api.travis-ci.org
```
![image](https://user-images.githubusercontent.com/1954436/28168135-c999bdfc-67e6-11e7-8e1f-b130b2a77e58.png)

`result = 0` - all is good.

Docs:
https://docs.travis-ci.com/api/#branches
https://docs.travis-ci.com/api/#jobs

**Comment by peter-gribanov** [comment]
Example failed job

https://travis-ci.org/symfony/symfony/jobs/253198333
https://api.travis-ci.org/jobs/253198333

![image](https://user-images.githubusercontent.com/1954436/28168668-7d58f9f6-67e8-11e7-8f1a-abee7a2e26d6.png)

**Comment by paulmelnikow** [comment]
Would someone like to work on a pull request?

**Comment by peter-gribanov** [comment]
Please note that the result for the same project will vary depending on the approach used.

For project: https://github.com/yiisoft/yii

[![PHP](https://img.shields.io/badge/PHP-%3E%3D5.1.0-blue.svg)](https://github.com/yiisoft/yii) by Packagist #1256 see [composer.json](https://github.com/yiisoft/yii/blob/master/composer.json#L76)
[![PHP](https://img.shields.io/badge/PHP-5.3%20--%207.1%2C%20HHVM-blue.svg)](https://github.com/yiisoft/yii) by Travis #1258 see [.travis.yml](https://github.com/yiisoft/yii/blob/master/.travis.yml#L4-L9)

**Comment by paulmelnikow** [comment]
Closed via #1256, #1258, and #1372. Thanks so much for all your work @peter-gribanov!

**Comment by vanderlee** [comment]
Much thanks for all the brilliant work on this one!
